# Sajjad Rahbri 

In [3]:
import numpy as np

array1 = np.random.randint(0,255,(1000,1000))
array2 = np.random.randint(0,255,(1000,1000))
result = np.zeros((1000,1000))

In [4]:
array1

array([[239, 247, 205, ...,  17, 237, 243],
       [123,  54,  16, ...,  63,  28, 250],
       [122, 176, 162, ...,  97, 170, 155],
       ...,
       [ 74, 157, 248, ...,  54,  39, 209],
       [230, 231,  96, ..., 107, 134,   3],
       [ 80,  54, 127, ...,  78,  31, 218]], dtype=int32)

In [5]:
array2

array([[121, 245, 208, ...,  71, 243,  41],
       [183, 146,  79, ..., 244, 115,  29],
       [223, 227,  22, ..., 178, 230, 231],
       ...,
       [158, 212, 163, ..., 130,  72, 240],
       [  5, 196,  96, ...,  64,  86,  65],
       [248, 205, 241, ...,  48,   7, 247]], dtype=int32)

## By Loop for 

In [10]:
%%time
column = 0
row = 0

for i1 in range(1000):
    for i2 in range(1000):
        arr1 = array1[i1][i2]
        arr2 = array2[i1][i2]

        if arr1 == arr2:
            result[i1,i2] = 225
        else: 
            result[i1,i2] = 0
        row += 1
    column += 1

CPU times: total: 141 ms
Wall time: 451 ms


In [11]:
result

array([[  0,   0,   0, ...,   0,   0,   0],
       [  0,   0,   0, ...,   0,   0,   0],
       [  0,   0,   0, ...,   0,   0,   0],
       ...,
       [  0,   0,   0, ...,   0,   0,   0],
       [  0,   0, 225, ...,   0,   0,   0],
       [  0,   0,   0, ...,   0,   0,   0]])

## By numpy method

In [12]:
%%time
result = np.where(array1 == array2,255,0)

CPU times: total: 0 ns
Wall time: 2.04 ms


In [14]:
result

array([[  0,   0,   0, ...,   0,   0,   0],
       [  0,   0,   0, ...,   0,   0,   0],
       [  0,   0,   0, ...,   0,   0,   0],
       ...,
       [  0,   0,   0, ...,   0,   0,   0],
       [  0,   0, 255, ...,   0,   0,   0],
       [  0,   0,   0, ...,   0,   0,   0]])

## result 

time in result1 (use for) : 451 ms

time in result1 (use np.where) : 2 ms